api/src/main/java/jakarta/data/metamodel/SortableAttribute.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023,2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.Sort;
+
+/**
+ * Represents a sortable entity attribute in the {@link StaticMetamodel}.
+ * Entity attribute types that are sortable include:
+ *
+ * <ul>
+ * <li>numeric attributes</li>
+ * <li>enum attributes</li>
+ * <li>time attributes</li>
+ * <li>boolean attributes</li>
+ * <li>{@link TextAttribute textual attributes}</li>
+ * </ul>
+ */
+public interface SortableAttribute extends Attribute {
```

Comment:
IMHO: by default, all attributes are sortable because the database defines this aspect, not the Java programming.

I mean, I don't have any scenario where the Java side will define if the attribute is sortable or not.

Reply:
Hrrrm. Good point. That's probably right. I mean, I bet we could find some sort of data store where binary data is not sortable, but I guess we need to model that in the API.